src/ControllerStatusIO.c

```diff
@@ -751,3 +763,86 @@ int Ros_Controller_GetActiveAlarmCodes(USHORT active_alarms[MAX_ALARM_COUNT + MA
+        if (Ros_mpGetRobotCalibrationData(i, &calibData) == OK)
+        {
+            Ros_Debug_BroadcastMsg("%s: file %d loaded OK", __func__, i);
+
+            //Take note at least one calibration file loaded successfully.
+            //
+            //NOTE: successful loading of a single calibration file will of
+            //course not mean all groups (that would benefit from it) are actually
+            //calibrated correctly. But this is the best we can do, as there is
+            //no requirement to calibrate every group to every other group, nor
+            //will this prevent MotoROS2 from successfully controlling motion.
+            //
+            //TF frame broadcasts for uncalibrated groups will however be incorrect,
+            //which is why posting an alarm in case there are no calibration files
+            //at all on a multigroup system is considered a good thing to do.
+            bOneCalibFileLoaded = TRUE;
+
+            if (calibData.s_rb.grp_no <= MP_R8_GID && //the slave is a robot
+                calibData.m_rb.grp_no <= MP_R8_GID) //the master is another robot's RF
+            {
+                int groupIndex = mpCtrlGrpId2GrpNo((MP_GRP_ID_TYPE)calibData.s_rb.grp_no);
+                MP_COORD* coord = &controller->ctrlGroups[groupIndex]->robotCalibrationToBaseFrame;
+                coord->x = calibData.pos_uow[0];
+                coord->y = calibData.pos_uow[1];
+                coord->z = calibData.pos_uow[2];
+                coord->rx = calibData.ang_uow[0];
+                coord->ry = calibData.ang_uow[1];
+                coord->rz = calibData.ang_uow[2];
+            }
+        }
+    }
+
+    return bOneCalibFileLoaded;
+}
+
+//TODO(gavanderhoorn): should be able to make this 'static' once Yaskawa-Global/motoros2#173
+//is implemented and we can unit test static functions
+BOOL Ros_Controller_ShouldWarnNoCalibDataLoaded(Controller const* controller, BOOL bCalibLoadedOk, BOOL bPublishTfEnabled)
+{
+    size_t numRobotsOrStations = 0;
+    for(size_t i=0; i < controller->numGroup; ++i)
+        if (Ros_CtrlGroup_IsRobot(controller->ctrlGroups[i]) || Ros_CtrlGroup_IsStation(controller->ctrlGroups[i]))
+            numRobotsOrStations += 1;
+
+    //It's OK if calibration did not load (for single-group systems fi, or systems
+    //that haven't been calibrated), but it might be prudent to post an alarm just
+    //in case IFF:
+    //
+    // - this is a multi-group system
+    // - there is at least one robot
+    // - there are groups other than base groups (so R1+R2, or R1+S1)
+    // - TF broadcast is enabled, and
+    // - none of the calibration files loaded
+    //
+    //in that case there could be Robot groups that have not been calibrated (to
+    //either other robots and/or Station groups) and this would make their TF
+    //broadcasts incorrect.
+    return (
+        //single-group systems don't have calibration files at all
+        controller->numGroup > 1 &&
```

Comment:
This line is probably not needed, since `numRobotsOrStations > 1` exists below

Reply:
Getting back to this: I mostly added the `numRobot` and `numGroup` check because of what we discussed in https://github.com/Yaskawa-Global/motoros2/pull/169#discussion_r1353202048:

@gavanderhoorn wrote:
> Would it be correct to say that any multi-group system with non-base-axis groups would fall into the `raise-alarm-if-no-calibration-files-loaded` category?
> 
> IIUC, base axes groups are always associated with robot groups, correct? So that leaves just station axes and proper robots.

If it's impossible to have just a station, and base axes are always associated with a robot group, then just counting robots and stations would indeed be enough.